concrete/src/Express/Association/Formatter/AbstractFormatter.php

```diff
@@ -33,10 +33,35 @@ public function getEntryDisplayName(AssociationControl $control, Entry $entry)
             } catch (\Exception $e) {
             }
         }
-        $targetEntity = $this->association->getTargetEntity();
-        $attribute = $targetEntity->getAttributes()[0];
+
+        $entity = $entry->getEntity();
+
+        /** @var \Concrete\Core\Entity\Attribute\Key\ExpressKey[] $attributes */
+        $attributes = $entity->getAttributes();
+
+        if ($keyId = $entity->getDisplayAttributeKey()) {
+
+            foreach ($attributes as $key) {
+                if ($key->getAttributeKeyID() == $keyId) {
+                    $attribute = $key;
+                    break;
+                }
+            }
+        } else {
+            foreach ($attributes as $attribute) {
+                if ($attribute->getAttributeType()->getAttributeTypeHandle() === 'text') {
```

Comment:
Worse than what the replaces? Ideally all entities will have a configured display attribute.